test/test_serde_full.py

```diff
@@ -0,0 +1,1623 @@
+from collections import OrderedDict
+import pytest
+import numpy
+import torch
+from functools import partial
+import traceback
+import io
+
+import syft
+from syft.serde import serde
+
+# Make dict of type codes
+CODE = OrderedDict()
+for cls, simplifier in serde.simplifiers.items():
+    CODE[cls] = simplifier[0]
+
+
+########################################################################
+# Functions that return list of serde samples in the following format:
```

Comment:
I agree on this point, but more because of `test_serde_simplify` which is not 100% clear because it uses indices of the tuple.
By the way, I see that when you define a `compare` function, you always do for 1 (custom_detailed_values_comparison_function ?) and the other one (custom_simplified_values_comparison_function ?) is always None, is this last one needed?

Reply:
@karlhigley Please take a look now?

Reply:
Thanks, that helps! Turns out I was misreading the examples the first time. 😅